crates/hyperdrive-math/src/short/open.rs

```diff
@@ -44,117 +70,197 @@ impl State {
+        // vault price:
+        //
+        // $$
+        // \text{base_proceeds} = (
+        //    \frac{c1 \cdot \Delta y}{c0 \cdot c}
+        //    + \frac{\Delta y \cdot \phi_f}{c} - \Delta z
+        // ) \cdot c
+        // $$
+        let base_proceeds = self
+            .calculate_short_proceeds_up(
+                bond_amount,
+                share_reserves_delta - curve_fee_shares,
+                open_vault_share_price,
+                close_vault_share_price,
+            )
+            .mul_up(self.vault_share_price());
+
+        Ok(base_proceeds)
     }
 
-    /// Calculates the derivative of the short deposit function with respect to the
-    /// short amount. This allows us to use Newton's method to approximate the
-    /// maximum short that a trader can open.
+    /// Calculates the derivative of the short deposit function with respect to
+    /// the short amount.
+    ///
+    /// This derivative allows us to use Newton's method to approximate the
+    /// maximum short that a trader can open. The share adjustment derivative is
+    /// a constant:
+    ///
+    /// $$
+    /// P^{\prime}_{\text{adj}}(\Delta y) = \tfrac{c_{1}}{c_{0} \cdot c} + \tfrac{\phi_{f}}{c}
+    /// $$
     ///
-    /// Using this, calculating $D'(\Delta y)$ is straightforward:
+    /// The curve fee dervative is given by:
     ///
     /// $$
-    /// D'(\Delta y) = \tfrac{c}{c_0} - (c \cdot P'(\Delta y) - \phi_{curve} \cdot (1 - p)) + \phi_{flat}
+    /// \Phi^{\prime}_{\text{c}}(\Delta y) = \phi_{c} \cdot (1 - p_0),
     /// $$
     ///
+    /// where $p_0$ is the opening (or initial) spot price. Using these and the
+    /// short principal derivative, we can calculate the open short derivative:
+    ///
     /// $$
-    /// P'(\Delta y) = \tfrac{1}{c} \cdot (y + \Delta y)^{-t_s} \cdot \left(\tfrac{\mu}{c} \cdot (k - (y + \Delta y)^{1 - t_s}) \right)^{\tfrac{t_s}{1 - t_s}}
+    /// D^{\prime}(\Delta y) =
+    ///\begin{cases}
+    ///    c \cdot \left(
+    ///      P^{\prime}_{\text{adj}}(\Delta y)
+    ///      - P^{\prime}_{\text{lp}}(\Delta y)
+    ///      + \Phi^{\prime}_{c,os}(\Delta y)
+    ///    \right),
+    ///    & \text{if }
+    ///      P_{\text{adj}} > P_{\text{lp}}(\Delta y) - \Phi_{c,os}(\Delta y) \\
+    ///    0, & \text{otherwise}
+    ///\end{cases}
     /// $$
     pub fn calculate_open_short_derivative(
         &self,
         bond_amount: FixedPoint,
         open_vault_share_price: FixedPoint,
         maybe_initial_spot_price: Option<FixedPoint>,
     ) -> Result<FixedPoint> {
+        // We're avoiding negative interest, so we will cap the close share
+        // price to be greater than or equal to the open share price. This
+        // will assume the max loss for the trader, some of which may be
+        // reimbursed upon closing.
+        let close_vault_share_price = open_vault_share_price.max(self.vault_share_price());
+
+        // Short circuit the derivative if the forward function returns 0.
```

Comment:
Nit: Mind if we stick to "derivative" and "function"? Forward and backward seem a little out of place here, but it's not a hill that I'm going to die on.

Reply:
yes definitely -- I agree that phrasing should be reserved for ML code. Including it here was autonomic